docs/architecture/adr-007-specialization-groups.md

```diff
@@ -0,0 +1,161 @@
+    Active() bool 
+
+    // functionality to execute for when a vote is cast in this election, here
+    // the vote field is anticipated to be marshalled into a vote type used 
+    // by an election. 
+    Vote(addr sdk.AccAddress, vote []byte) 
+
+    // here lies all functionality to authenticate and execute changes for
+    // when a member accepts being elected
+    Accept(sdk.AccAddress) 
+
+    // Register a revoker object
+    RegisterRevoker(Revoker)
+
+    // No more revokers may be registered after this function is called
+    SealRevokers()
+
+    // register hooks to call when an election actions occur
+    RegisterHooks(ElectionatorHooks) 
+
+    // query for the current winner(s) of this election based on arbitrary
+    // election ruleset
+    QueryWinners() []sdk.AccAddress 
+
+    // query metadata for an address in the election this 
+    // could include for example position that an address
+    // is being elected for within a group   
+    // 
+    // this metadata may be directly related to 
+    // voting information and/or privileges enabled
+    // to members within a group. 
+    QueryMetadata(sdk.AccAddress) []byte
+}
+
+type ElectionatorHooks interface {
+    VoteCast(addr sdk.AccAddress, vote []byte)
+    MemberAccepted(addr sdk.AccAddress)
+    MemberRevoked(addr sdk.AccAddress, cause []byte)
+}
+
+// Revoker defines the function required for an membership revocation rule-set
+// used by a specialization group. This could be used to create self revoking,
+// and evidence based revoking, etc. Revokers types may be created and
+// reused for different election types. 
+// 
+// When revoking the "cause" bytes may be arbitrarily marshalled into evidence,
+// memos, etc.
+type Revoker interface {
+    RevokeName() string      // identifier for this revoker type 
+    RevokeMember(addr sdk.AccAddress, cause []byte) (successful bool)
+}
+```
+
+Certain level of commonality likely exists between the existing code within
+`x/governance` and required functionality of elections. This common
+functionality should be abstracted during implementation. Similarly for each 
+vote implementation client CLI/REST functionality should be abstracted
+to be reused for multiple elections. 
+
+The specialization group abstraction firstly extends the `Electionator`
+but also further defines traits of the group. 
+
+``` golang
+type SpecializationGroup interface {
+    Electionator 
+    GetName() string
+    GetDescription() string 
+
+    // general soft contract the group is expected
+    // to fulfill with the greater community
+    GetContract() string
+
+    // messages which can be executed by the members of the group
+	Handler(ctx sdk.Context, msg sdk.Msg) sdk.Result
+
+    // logic to be executed at endblock, this may for instance
+    // include payment of a stipend to the group members
+    // for participation in the security group.   
+	EndBlocker(ctx sdk.Context)
+}
+```
+
+## Status
+
+> Proposed
+
+## Consequences
```

Comment:
it'd be helpful to mention if this requires to do further changes to the current governance module (if any)

Reply:
added ` - improve abstractions in `x/gov/` such that they can be used with specialization groups` to Positives though